src/main/java/com/saucelabs/ci/sauceconnect/AbstractSauceTunnelManager.java

```diff
@@ -697,6 +850,23 @@ protected abstract String[] generateSauceConnectArgs(
 
   protected abstract String[] addExtraInfo(String[] args);
 
+  /**
+   * Returns the arguments to be used to launch Sauce Connect
+   *
+   * @param args the initial Sauce Connect command line args
+   * @param username name of the user which launched Sauce Connect
+   * @param apiKey the access key for the Sauce user
+   * @param options command line args specified by the user
+   * @return String array representing the command line args to be used to launch Sauce Connect
+   * @deprecated Use {@link #generateSauceConnectArgs(String[], String, String, String)} instead
+   */
+  @Deprecated
+  protected abstract String[] generateSauceConnectArgsLegacy(
```

Comment:
It looks like `generateSauceConnectArgsLegacy` and `addExtraInfoLegacy` are not used in `AbstractSauceTunnelManager` at all. They can defined in `SauceConnectManager` only.

Reply:
Fixed in f8c3517